src/analyses/ai.h

```diff
@@ -397,24 +435,31 @@ class ait:public ai_baset
     return util_make_unique<domainT>(it->second);
   }
 
+  /// Remove all analysis results.
   void clear() override
   {
     state_map.clear();
     ai_baset::clear();
   }
 
 protected:
+  /// Map from locations to domain elements, for the results of a static analysis.
   typedef std::
     unordered_map<locationt, domainT, const_target_hash, pointee_address_equalt>
       state_mapt;
   state_mapt state_map;
 
+  /// Look up the analysis state for a given location, instantiating a new state if required.
+  /// Used internally by the analysis.
   // this one creates states, if need be
   virtual statet &get_state(locationt l) override
   {
     return state_map[l]; // calls default constructor
   }
 
+  /// Look up the analysis state for a given location, throwing an exception if no state is
+  /// known.
+  /// Used internally by the analysis.
```

Comment:
Remove the existing documentation, which is superseded by your new documentation